neard/tests/node_cluster.rs

```diff
@@ -0,0 +1,104 @@
+use futures::future;
+
+use near_actix_test_utils::{run_actix_until_stop, spawn_interruptible};
+use near_client::{ClientActor, ViewClientActor};
+use near_primitives::types::{BlockHeight, BlockHeightDelta, NumSeats, NumShards};
+use testlib::{start_nodes, test_helpers::heavy_test};
+
+#[derive(Debug, Default)]
+pub struct NodeCluster {
+    dirs: Vec<tempfile::TempDir>,
+    is_heavy: bool,
+    num_shards: Option<NumShards>,
+    num_validator_seats: Option<NumSeats>,
+    num_lightclient: Option<usize>,
+    epoch_length: Option<BlockHeightDelta>,
+    genesis_height: Option<BlockHeight>,
+}
+
+macro_rules! add_mut_helpers {
+    ($($method:ident($self:ident $(,$arg:ident:$type:ty)?) => $expr:expr),+) => {
+        $(
+            pub fn $method(mut $self$(, $arg: $type)?) -> Self {
+                $expr;
+                $self
+            }
+        )+
+    }
+}
+
+impl NodeCluster {
+    pub fn new<F: Fn(usize) -> String>(capacity: usize, gen_dirname: F) -> Self {
+        Self {
+            dirs: (0..capacity)
+                .map(|index| {
+                    tempfile::Builder::new().prefix(&gen_dirname(index)).tempdir().unwrap()
+                })
+                .collect(),
+            ..Default::default()
+        }
+    }
+
+    add_mut_helpers! {
+        set_heavy_test(self) => self.is_heavy = true,
+        set_shards(self, n: NumShards) => self.num_shards = Some(n),
```

Comment:
set_shards is too broad naming. Let’s go with set_shards_number.

I also propose to rename set_validator_seats_count

Reply:
here's a couple of probable names, ticked are the ones I think I prefer, what do you think?

- ∞
  - [X] `set_shard_count`
  - [ ] `set_num_shards`
  - [ ] `set_shards_number`
- ∞
  - [X] `set_validator_seat_count`
  - [ ] `set_validator_seats_count`
  - [ ] `set_num_validator_seats`

Reply:
Then again, `set_num_` prefixed names seems to be the most intuitive.